src/main/kotlin/pw/forst/tools/katlib/IterableExtensions.kt

```diff
@@ -497,3 +509,57 @@ inline fun <TItem> Iterable<TItem>.itemsToString(
     }
     return sb.toString().restrictLengthWithEllipsis(totalLength)
 }
+
+/**
+ * Returns a single list of all not null elements yielded from results of [transform]
+ * function being invoked on each element of original collection.
+ */
+fun <T, R> Iterable<T>.flatMapIndexedNotNull(transform: (index: Int, T) -> Iterable<R>?): List<R> {
+    return flatMapIndexedTo(ArrayList(), transform)
```

Comment:
as the `flatMapIndexedTo` is inlined, this one can be too, please inline it, thanks!